pkg/mlsstore/store.go

```diff
@@ -125,6 +121,59 @@ func (s *Store) ConsumeKeyPackages(ctx context.Context, installationIds []string
 	return keyPackages, nil
 }
 
+func (s *Store) GetIdentityUpdates(ctx context.Context, walletAddresses []string, startTimeNs int64) (map[string]IdentityUpdateList, error) {
+	updated := make([]*Installation, 0)
+	// Find all installations that were changed since the startTimeNs
+	err := s.db.NewSelect().
+		Model(&updated).
+		Where("wallet_address IN (?)", bun.In(walletAddresses)).
+		WhereGroup(" AND ", func(q *bun.SelectQuery) *bun.SelectQuery {
+			return q.Where("created_at > ?", startTimeNs).WhereOr("revoked_at > ?", startTimeNs)
+		}).
+		Order("created_at ASC").
+		Scan(ctx)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// The returned list is only partially sorted
+	out := make(map[string]IdentityUpdateList)
+	for _, installation := range updated {
+		if installation.CreatedAt > startTimeNs {
+			out[installation.WalletAddress] = append(out[installation.WalletAddress], IdentityUpdate{
+				Kind:           Create,
+				InstallationId: installation.ID,
+				TimestampNs:    uint64(installation.CreatedAt),
+			})
+		}
+		if installation.RevokedAt != nil && *installation.RevokedAt > startTimeNs {
+			out[installation.WalletAddress] = append(out[installation.WalletAddress], IdentityUpdate{
+				Kind:           Revoke,
+				InstallationId: installation.ID,
+				TimestampNs:    uint64(*installation.RevokedAt),
+			})
+		}
+	}
+	// Sort the updates by timestamp now that the full list is assembled
+	for _, updates := range out {
+		sort.Sort(updates)
+	}
+
+	return out, nil
+}
+
+func (s *Store) RevokeInstallation(ctx context.Context, installationId string) error {
+	_, err := s.db.NewUpdate().
+		Model(&Installation{}).
+		Set("revoked_at = ?", nowNs()).
+		Where("id = ?", installationId).
+		Where("revoked_at IS NULL").
```

Comment:
Could make sense to use `not_revoked` here too

Reply:
Since it's scoping on installation ID first it's probably not a big deal to just use revoked_at IS NULL tbh